Mathlib/GroupTheory/PresentedGroup.lean

```diff
@@ -94,6 +94,27 @@ theorem ext {φ ψ : PresentedGroup rels →* G} (hx : ∀ (x : α), φ (.of x)
   ext
   apply hx
 
+variable {β : Type*}
+
+/-- Presented groups of isomorphic types are isomorphic. -/
+def equivPresentedGroup (rels : Set (FreeGroup α)) (e : α ≃ β) :
+    PresentedGroup rels ≃* PresentedGroup ((FreeGroup.freeGroupCongr e) '' rels) :=
+  QuotientGroup.congr (Subgroup.normalClosure rels)
+    (Subgroup.normalClosure ((FreeGroup.freeGroupCongr e)'' rels)) (FreeGroup.freeGroupCongr e)
+    (Subgroup.map_normalClosure rels (FreeGroup.freeGroupCongr e).toMonoidHom
+      (FreeGroup.freeGroupCongr e).surjective)
+
+theorem equivPresentedGroup_apply_of (x : α) (rels : Set (FreeGroup α)) (e : α ≃ β) :
+    equivPresentedGroup rels e (PresentedGroup.of x) =
+    PresentedGroup.of (rels := (FreeGroup.freeGroupCongr e) '' rels) (e x) := by
+  rfl
+
+theorem equivPresentedGroup_symm_apply_of (x : α) (rels : Set (FreeGroup α)) (e : α ≃ β) :
+    (equivPresentedGroup rels e).symm
+      (PresentedGroup.of (rels := ((FreeGroup.freeGroupCongr e) '' rels)) (e x)) =
+    PresentedGroup.of (rels := rels) (e.symm (e x)) := by
+  rfl
```

Comment:
I'm surprised these `rels` hints are needed. We can remove a couple of them to tidy it up. In any case, for the second theorem we want the `e.symm (e x)` to be simplified to `x`. We can also make it more general when rewriting left-to-right by having it be in terms of `(x : β)`. Here:
```suggestion
theorem equivPresentedGroup_apply_of (x : α) (rels : Set (FreeGroup α)) (e : α ≃ β) :
    equivPresentedGroup rels e (PresentedGroup.of x) =
      PresentedGroup.of (rels := FreeGroup.freeGroupCongr e '' rels) (e x) := by
  rfl

theorem equivPresentedGroup_symm_apply_of (x : β) (rels : Set (FreeGroup α)) (e : α ≃ β) :
    (equivPresentedGroup rels e).symm (PresentedGroup.of x) =
      PresentedGroup.of (rels := rels) (e.symm x) := by
  rw [MulEquiv.symm_apply_eq, equivPresentedGroup_apply_of, Equiv.apply_symm_apply]
```

Reply:
I just edited that. Turns out the second one is true by `rfl` rather than a more involved `rw` proof.

By the way, making the proof by `rfl` instead of `by rfl` makes it available to `dsimp`.